third_party/nvidia/hopper/lib/Transforms/WarpSpecialization/WSTaskPartition.cpp

```diff
@@ -0,0 +1,163 @@
+#include "Utility.h"
+#include "mlir/Analysis/SliceAnalysis.h"
+#include "mlir/Pass/Pass.h"
+#include "mlir/Pass/PassManager.h"
+#include "mlir/Transforms/Passes.h"
+#include "nvidia/hopper/include/Transforms/Passes.h"
+#include "triton/Dialect/TritonGPU/IR/Dialect.h"
+#include "triton/Dialect/TritonGPU/Transforms/Utility.h"
+#include "triton/Dialect/TritonNvidiaGPU/IR/Dialect.h"
+
+#define DEBUG_TYPE "nvgpu-ws-task-partition"
+#define DBGS() (llvm::dbgs() << "[" DEBUG_TYPE "]: ")
+#define LDBG(X) LLVM_DEBUG(DBGS() << X << "\n")
+
+namespace tt = mlir::triton;
+namespace ttg = mlir::triton::gpu;
+namespace ttng = ::mlir::triton::nvidia_gpu;
+namespace mlir {
+
+struct TaskSchedule {
```

Comment:
nit: everything but the  pass definition should be in an anonymous namespace to prevent leaking the linkage name

Reply:
Thanks for the detailed review and all suggestions! Addressed all of them except for this one.

The function `doTaskPartition` is actually used in another file, i.e, `WarpSpecialization.cpp` and I need to expose it. Let me know any suggestions. Thanks!